src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -101,7 +104,24 @@ private static bool IsMSBuildAssembly(Assembly assembly)
 
         private static bool IsMSBuildAssembly(AssemblyName assemblyName)
         {
-            return s_msBuildAssemblies.Contains(assemblyName.Name, StringComparer.OrdinalIgnoreCase);
+            if (!s_msBuildAssemblies.Contains(assemblyName.Name, StringComparer.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            var publicKeyToken = assemblyName.GetPublicKeyToken();
+            if (publicKeyToken == null || publicKeyToken.Length == 0)
+            {
+                return false;
+            }
+
+            var sb = new StringBuilder();
```

Comment:
```C#
new StringBuilder(MSBuildPublicKeyToken.Length);
```

to avoid allocations. Though I guess there'll be garbage strings for each byte later so it probably doesn't matter.